### PR TITLE
migrate to hardhat-chai-matchers and hardhat-toolbox

### PR DESCRIPTION
**Pull request by ahmedali8**
Resolves issue: [#112](https://github.com/paulrberg/solidity-template/issues/112)

### CODE REVIEW

**Comment by ahmedali8** [comment]
Resolves issue: #120

**Comment by ahmedali8** [comment]
Hey @paulrberg. Conflicts resolved!

File: test/greeter/Greeter.ts [hunk 1]
```diff
@@ -1,24 +1,38 @@
+import { loadFixture } from "@nomicfoundation/hardhat-network-helpers";
 import type { SignerWithAddress } from "@nomiclabs/hardhat-ethers/dist/src/signer-with-address";
-import { artifacts, ethers, waffle } from "hardhat";
-import type { Artifact } from "hardhat/types";
+import { ethers } from "hardhat";
 
+import { Greeter__factory } from "../../src/types";
```
Comment (paulrberg) [inline test/greeter/Greeter.ts:5]: This should use "import type" and be imported from "../../src/types/factories/Greeter__factory".

**Comment by paulrberg** [comment]
Hey @ahmedali8, just submitted my review. Here's a bit of additional feedback, if I may:

1. It looks like you [merged](https://github.com/paulrberg/solidity-template/pull/119/commits/3d61198ee27f55c014df05f2f068a077e3eed7e4) my `main` branch instead of rebasing - in the majority of cases like this when you submit a PR to a repo and that repo has updated its `main` tip, it's better to use [git rebase](https://www.atlassian.com/git/tutorials/merging-vs-rebasing) instead. Rebasing keeps the historical commits in chronological order and it also tends to create fewer git conflicts.
2. It looks like you didn't give me permission to contribute to your branch - this is generally useful so that I don't have to request changes from your for every minor thing that should be changed that I could change myself.
3. It's clearer for historical purposes to create a bespoke branch name on your fork, e.g. in this case something like `refactor/hardhat-toolbox` would have been nice.

**Comment by ahmedali8** [comment]
image

hey @paulrberg, thanks alot for the guidance. 
1. I'll make sure to follow this for future PRs as I didn't know this previously.
2. given you permission already since the beginning, screenshot attached.
3. should I close this and create a new PR or what should I do please guide?

File: test/greeter/Greeter.ts (hunk 1 above)
Comment (ahmedali8) [reply test/greeter/Greeter.ts]: done

**Comment by paulrberg** [comment]
> given you permission already since the beginning, screenshot attached.

Oh, shoot. Sorry then! My bad.

**Pull request closed without merging**